amalgamation.sh

```diff
@@ -173,7 +173,7 @@ SINGLEHDR=$SCRIPTPATH/singleheader
 echo "Copying files to $SCRIPTPATH/singleheader "
 mkdir -p $SINGLEHDR
 echo "c++ -O3 -std=c++17 -pthread -o ${CPPBIN} ${DEMOCPP}  && ./${CPPBIN} ../jsonexamples/twitter.json ../jsonexamples/amazon_cellphones.ndjson" > $SINGLEHDR/README.md
-cp ${AMAL_C} ${AMAL_H}  ${DEMOCPP} $SINGLEHDR
+cp --remove-destination --preserve=all ${AMAL_C} ${AMAL_H}  ${DEMOCPP} $SINGLEHDR
```

Comment:
What is the purpose of remove-destination here, for my edification?

Reply:
I do not think that these flags (--remove-destination --preserve=all) are POSIX. They won't work on macOS and other system. I suspect that they are GNU-specific.

This certainly breaks the script on macOS.

Reply:
problem is/was that 

> > And there were places where members were not initialized and complier complained about that.
> 
> Where? Maybe I missed them. For the cases that I looked at, there was initialization.

For example:
 /** @private Structural indices passed from stage 1 to stage 2 */
  std::unique_ptr<uint32_t[]> structural_indexes;

If you want to have whole list, try to compile with -Wellc++. ;-)

Reply:
@ostri

> std::unique_ptr<uint32_t[]> structural_indexes;

This is not a valid error. The structural_indexes class instance will get initialized with its default constructor. This is guaranteed by the standard.

I know that -Weffc++ throws a ton of warnings, but most are invalid.

Reply:
I think "--remove-destination --preserve=all" is not portable and it seems unrelated to this PR. If you have an issue with the script, please let us discuss it elsewhere.